services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplAutowiredTest.java

```diff
@@ -106,54 +118,238 @@ public void tearDown() throws Exception {
 	@Test
-	public void testAuthorizationRoundTrip() throws Exception {		
+	public void testAuthorizationCodeRoundTrip() throws Exception {
 
 		// Verify the client
 		oauthClient = oauthClientManager.updateOpenIDConnectClientVerifiedStatus(adminUserInfo, oauthClient.getClient_id(), oauthClient.getEtag(), true);
-		
+
 		OIDCAuthorizationRequest authorizationRequest = new OIDCAuthorizationRequest();
 		authorizationRequest.setClientId(oauthClient.getClient_id());
 		authorizationRequest.setRedirectUri(oauthClient.getRedirect_uris().get(0));
-		authorizationRequest.setScope(OAuthScope.openid.name());
+		authorizationRequest.setScope(OAuthScope.openid.name() + " " + OAuthScope.offline_access.name());
 		authorizationRequest.setResponseType(OAuthResponseType.code);
-		authorizationRequest.setClaims("{\"id_token\":{\"team\":{\"values\":[\"2\"]}},\"userinfo\":{\"team\":{\"values\":[\"2\"]}}}");
-				
+		Map<String, OIDCClaimsRequestDetails> claimsToRequest = new HashMap<>();
+		OIDCClaimsRequestDetails teamClaimRequestDetails = new OIDCClaimsRequestDetails();
+		teamClaimRequestDetails.setValues(Collections.singletonList("2"));
+		claimsToRequest.put(OIDCClaimName.team.name(), teamClaimRequestDetails);
+		OIDCClaimsRequest claimsRequest = new OIDCClaimsRequest();
+		claimsRequest.setId_token(claimsToRequest);
+		claimsRequest.setUserinfo(claimsToRequest);
+		authorizationRequest.setClaims(claimsRequest);
+
 		// method under test
-		OIDCAuthorizationRequestDescription description = 
+		OIDCAuthorizationRequestDescription description =
 				openIDConnectManager.getAuthenticationRequestDescription(authorizationRequest);
-		
+
 		assertNotNull(description);
-		
+
 		// method under test
 		OAuthAuthorizationResponse authResponse = openIDConnectManager.
 				authorizeClient(userInfo, authorizationRequest);
-		
+
 		assertNotNull(authResponse.getAccess_code());
-		
+
 		// method under test
-		OIDCTokenResponse tokenResponse = 
-				openIDConnectManager.getAccessToken(authResponse.getAccess_code(), 
+		OIDCTokenResponse tokenResponse =
+				openIDConnectManager.getTokenResponseWithAuthorizationCode(authResponse.getAccess_code(),
 						oauthClient.getClient_id(), oauthClient.getRedirect_uris().get(0), OAUTH_ENDPOINT);
-		
-		
+
+
 		assertNotNull(tokenResponse.getAccess_token());
 		assertNotNull(tokenResponse.getId_token());
-		
+		assertNotNull(tokenResponse.getRefresh_token());
+
 		oidcTokenHelper.validateJWT(tokenResponse.getId_token());
-		
+
+		// method under test
+		JWTWrapper oidcUserInfo = (JWTWrapper) openIDConnectManager.getUserInfo(tokenResponse.getAccess_token(), OAUTH_ENDPOINT);
+
+		oidcTokenHelper.validateJWT(oidcUserInfo.getJwt());
+
+	}
+
+	@Test
+	public void testRefreshTokenRoundTrip() throws Exception {
+		// Verify the client
+		oauthClient = oauthClientManager.updateOpenIDConnectClientVerifiedStatus(adminUserInfo, oauthClient.getClient_id(), oauthClient.getEtag(), true);
```

Comment:
it looks like some of the same set up is used >=4 times.  Please factor out the repeated code.